packages/components/src/mobile/segmented-control/index.native.js

```diff
@@ -149,6 +149,17 @@ const SegmentedControls = ( {
 		<View style={ styles.row }>
 			<View style={ styles.flex }>{ addonLeft }</View>
 			<View style={ [ containerStyle, isIOS && styles.containerIOS ] }>
+				<Animated.View
+					style={ [
+						{
+							width,
+							left: positionAnimationValue,
+							height,
+						},
+						selectedStyle,
+						outlineStyle,
+					] }
+				/>
```

Comment:
I had to change the rendering order after upgrading `@react-navigation/stack` to version `6.3.5` (https://github.com/WordPress/gutenberg/pull/51303/commits/b5c02fd4936dd2dfd63cc73fd0ed807871ed05f0). Otherwise, the animated component was being rendered over the segments despite the `z-index` value.

<img src=https://github.com/WordPress/gutenberg/assets/14905380/59998f5b-de88-4c12-b3a7-cb455c6646d3 width=300>